scripts/scripts.js

```diff
@@ -396,6 +396,101 @@ async function loadEager(doc) {
   }
 }
 
+function videoEventTracking(myPlayer, videoTagId) {
+  let eventName = 'video_started';
+  let progress = [];
+
+  myPlayer.on('timeupdate', (event) => {
+    const videoPercentage = (myPlayer.currentTime() * 100) / myPlayer.duration();
+
+    switch (true) {
+      case videoPercentage === 100:
+        eventName = 'video_completed';
+        break;
+      case videoPercentage > 90:
+        eventName = 'video_reached_90_pct';
+        break;
+      case videoPercentage > 75:
+        eventName = 'video_reached_75_pct';
+        break;
+      case videoPercentage > 50:
+        eventName = 'video_reached_50_pct';
+        break;
+      case videoPercentage > 25:
+        eventName = 'video_reached_25_pct';
+        break;
+      default:
+        eventName = 'video_started';
+    }
+
+    if (!progress.includes(eventName)) {
+      progress.push(eventName);
+
+      const videoProgress = {
+        name: eventName,
+        digitalData: {
+          videoTracking: {
+            id: videoTagId,
+            name: myPlayer.mediainfo.name,
+          },
+        },
+        event,
+      };
+
+      if (eventName === 'video_completed') {
+        progress = [];
+      }
+
+      window.appEventData.push(videoProgress);
+    }
+  });
+}
+
+function socialShareTracking(doc) {
```

Comment:
I think this would make sense to be part of the social-share.js as it is very specific for that block, so basically block code.

Reply:
I had considered this, also the video tracking code might belong to embed/brightcove block. Once the brightcove is merged I will move the code there.